components/servo/lib.rs

```diff
@@ -961,8 +986,36 @@ impl Servo {
         if self.compositor.borrow().shutdown_state != ShutdownState::FinishedShuttingDown {
             self.compositor.borrow_mut().perform_updates();
         } else {
-            self.messages_for_embedder.push(EmbedderMsg::Shutdown);
+            self.messages_for_embedder
+                .borrow_mut()
+                .push(EmbedderMsg::Shutdown);
+        }
+    }
+
+    /// Spin the Servo event loop, which:
+    ///
+    ///   - Performs updates in the compositor, such as queued pinch zoom events
+    ///   - Runs delebgate methods on all `WebView`s and `Servo` itself
+    ///   - Maybe update the rendered compositor output, but *without* swapping buffers.
```

Comment:
What I see is compositor "receive messages first -> call delegate methods -> perform update." I think it should either be more transparent about what it exactly calls or update the compositor's method names to reflect what we want to explain here.

Reply:
Agreed. We actually plan to update the compositor names very soon to make it a lot clearer what they do.